### DISCUSSION TITLE
Using Renovate Config Repo to centralize the configuration of my projects

### DISCUSSION THREAD

**Discussion by joedayz** [renovatebot/renovate#17242] [Request Help]
Hi Guys:

I created a new Repo for this: 

github.com/quadimai/Quadim-Renovate-Config with only 2 files:
- renovate.json

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>quadimai/Quadim-Renovate-Config"
  ]
}
```

- default.json

```
{
    "extends": [
      "config:base"
    ],
    "labels": [
      "renovate"
    ],
    "pin": {
      "automerge": true
    },
    "packageRules": [
      {
        "matchUpdateTypes": ["minor", "patch", "digest"],
        "matchPackagePatterns": ["^quadim-typelib", "^quadim-springservicebaselib", "^aws-java-sdk", "^aws-sdk-go"],
        "automerge": true,
        "enabled": true,
        "major": {
          "automerge": false
        }          
      }
    ],
    "platformAutomerge": true
  }
```

I configure my others repos  with their configuration, 

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>quadimai/Quadim-Renovate-Config"
  ]
}

```

I am searching automerge to only it dependencies: "matchPackagePatterns": ["^quadim-typelib", "^quadim-springservicebaselib", "^aws-java-sdk", "^aws-sdk-go"]  and  "matchUpdateTypes": ["minor", "patch", "digest"]

Is it enough, or do I need to add another configuration in the repo?

Thanks for your support in advance.

**Comment by joedayz** [comment]
@rarkins  The automerge with its configuration is failing.

I received the PR with it message in one of my repos:

🚦 Automerge: Disabled by config. Please merge this manually once you are satisfied.

What is missing?, please, help me.

Jose

**Reply by rarkins** [reply]
for a start, remove this:

```
        "major": {
          "automerge": false
        }  
```

This is invalid config, may be causing a problem

**Reply by joedayz** [reply]
Fixed. The configuration is now:

```
{
  "extends": [
    "config:base"
  ],
  "labels": [
    "renovate"
  ],
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "digest"],
      "matchPackagePatterns": ["^quadim-typelib", "^quadim-springservicebaselib", "^aws-java-sdk", "^aws-sdk-go"],
      "automerge": true,
      "enabled": true       
    }
  ]
}
```

**Reply by rarkins** [reply]
Please take a screenshot of the PR you expect to be automerging but which isn't

**Reply by joedayz** [reply]
<img width="1560" alt="Screen Shot 2022-08-17 at 14 40 27" src="https://user-images.githubusercontent.com/1118135/185228084-310a7197-8b34-4616-8f4f-acb8e06ef454.png">

Sure. This was the last PR.

**Reply by viceice** [reply]
check the package name in PR, your pattern should match that value

**Reply by rarkins** [reply]
Here is the package prefix you should be matching: 
![image](https://user-images.githubusercontent.com/6311784/185230985-99b0a11f-95be-4d00-9a9b-9b0b6df6ed68.png)

**Reply by joedayz** [reply]
I did it change.  I am going to generate a new patch release and verify if the automerge works. Thanks.

```
{
  "extends": [
    "config:base"
  ],
  "labels": [
    "renovate"
  ],
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "digest"],
      "matchPackagePatterns": [
        "^ai.quadim.api:quadim", 
        "^github.com/aws/aws-sdk-go",
        "^com.amazonaws:aws-*"
      ],
      "automerge": true,
      "enabled": true       
    }
  ]
}
```

Note: I added two patterns more.

**Reply by viceice** [reply]
remove the enabled, it's the default value

**Reply by joedayz** [reply]
The automerge failed

<img width="813" alt="Screen Shot 2022-08-18 at 00 47 53" src="https://user-images.githubusercontent.com/1118135/185303564-110d64ea-6c29-405b-a339-c76a12abe4cb.png">

I see it text in the log:
`DEBUG: Created Pull Request #79(branch="renovate/quadim-springservicebaselib-version")
DEBUG: PR is configured for automerge(branch="renovate/quadim-springservicebaselib-version")
DEBUG: Checking schedule(at any time, null)(branch="renovate/quadim-springservicebaselib-version")
DEBUG: No schedule defined(branch="renovate/quadim-springservicebaselib-version")
DEBUG: resolveBranchStatus(branchName=renovate/quadim-springservicebaselib-version, ignoreTests=false)(branch="renovate/quadim-springservicebaselib-version")
DEBUG: getBranchStatus(renovate/quadim-springservicebaselib-version)(branch="renovate/quadim-springservicebaselib-version")
DEBUG: branch status check result(branch="renovate/quadim-springservicebaselib-version")
{
  "state": "pending",
  "statuses": []
}
DEBUG: No check runs found(branch="renovate/quadim-springservicebaselib-version")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
DEBUG: Branch status yellow(branch="renovate/quadim-springservicebaselib-version")
DEBUG: PR is not ready for merge (branch status is yellow)(branch="renovate/quadim-springservicebaselib-version")
DEBUG: getBranchPr(renovate/quadim-springservicebaselib-version)
DEBUG: findPr(renovate/quadim-springservicebaselib-version, undefined, open)
DEBUG: Found PR #79
DEBUG: Ensuring Dependency Dashboard
DEBUG: ensureIssue(Dependency Dashboard)
DEBUG: Patching issue
DEBUG: Issue updated
DEBUG: Removing any stale branches
DEBUG: config.repoIsOnboarded=true
DEBUG: Branch lists
{
  "branchList": [
    "renovate/quadim-springservicebaselib-version"
  ],
  "renovateBranches": [
    "renovate/quadim-springservicebaselib-version"
  ]
}`

I  created a branch rule. I think that it is blocking the PR:

<img width="1636" alt="2" src="https://user-images.githubusercontent.com/1118135/185303841-1cc82997-30fd-4bf7-a929-4e0a0d2d9923.png">

Should I delete it Branch protection rule?

I have it configuration in General:
<img width="1580" alt="1" src="https://user-images.githubusercontent.com/1118135/185304118-48ddfece-4e74-4ac9-b6cf-dd65b68bb8dc.png">

Should I have it configuration?

Please, confirm if I am ok or not.

Thanks in advance.

**Reply by viceice** [reply]
you need to run some kind of CI job for automerge, eg GitHub actions

**Reply by joedayz** [reply]
The mail has now a new message: Automerge: enabled
<img width="1226" alt="4" src="https://user-images.githubusercontent.com/1118135/185306658-4a425766-80c4-4255-8383-e9a06827ae7d.png">

In the log it shows: DEBUG: PR is configured for automerge but DEBUG: Branch status yellow(branch="renovate/quadim-springservicebaselib-version")
DEBUG: PR is not ready for merge (branch status is yellow)(branch="renovate/quadim-springservicebaselib-version")

<img width="1600" alt="3" src="https://user-images.githubusercontent.com/1118135/185306345-6c729511-f94e-43a8-80c2-9c78e4d9ce41.png">

Then according to your last message, I need to add the last step: configure an automerge with GitHub action.

Exists it in the documentation?

**Comment by rarkins** [comment]
Look up ignoreTests

I am now locking this thread as your original config problem has now been resolved
